gramps/gen/filters/optimizer.py

```diff
@@ -0,0 +1,137 @@
+#
+# Gramps - a GTK+/GNOME based genealogy program
+#
+# Copyright (C) 2024  Doug Blank <doug.blank@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+import logging
+
+LOG = logging.getLogger(".filter.optimizer")
+
+
+def intersection(sets):
+    if sets:
+        result = sets[0]
+        for s in sets[1:]:
+            result = result.intersection(s)
+        return result
+    else:
+        return set()
+
+
+def union(sets):
+    if sets:
+        result = sets[0]
+        for s in sets[1:]:
+            result = result.union(s)
+        return result
+    else:
+        return set()
```

Comment:
```suggestion
    return set()
```
Unnecessary else.

Reply:
I think it is more explicit, and "Explicit is better than implicit."